### PR TITLE
perf: avoid unnecessarily creating new list on every frame

### PR DESCRIPTION
**Pull request by ignatz**
… every time.

In my profiles this contributed sometimes several percent. The effect is probably also exacerbated by many layers. Subjectively this felt like a noticeable impact if I trust my noisy environment.

### CODE REVIEW

**Review by TesteurManiak — APPROVED** [review]
LGTM! Waiting for a second approval before merging.

**Review by JaffaKetchup — APPROVED** [review]
https://github.com/fleaflet/flutter_map/assets/58115698/3131a04d-d10d-458d-89f1-c5081d317d3a

**Pull request merged**
